crates/uv-resolver/src/resolver/mod.rs

```diff
@@ -267,18 +267,25 @@ impl<Provider: ResolverProvider, InstalledPackages: InstalledPackagesProvider>
         let solver = state.clone();
         let index_locations = provider.index_locations().clone();
         let (tx, rx) = oneshot::channel();
-        thread::Builder::new()
+        let handle = thread::Builder::new()
             .name("uv-resolver".into())
             .spawn(move || {
                 let result = solver.solve(index_locations, request_sink);
-                tx.send(result).unwrap();
+                tx.send(result)
```

Comment:
I think ignoring the send error here without the `join` would be an equivalent fix.. but I don't see how this could be causing 401s. 

```suggestion
                let _ = tx.send(result);
```

Reply:
I don't have a strong feelings. Also seems fine to join and check for an error? Perhaps vaguely better in case something changes in the future?

Reply:
I don't think the join is really meaningful because a send error implies that the main thread failed already, so it should never even reach the join. We shouldn't really be calling `join` anyways because it's blocking, that's why we use the channel for communication.

Reply:
Alright.